### PR TITLE
Added a way to disable validation on save method

### PR DESCRIPTION
**Pull request by D1ceWard**
This PR add an named argument to `#save` method which allow to disable model validation, example:

```crystal
class Example < Granite::Base
  ...
  column name : String?
  
  validate :name, "Name cannot be blank" do |parent|
    !parent.name.to_s.blank?
  end
  ...
end

new_instance = Example.new
new_instance.name = ""
new_instance.save(validate: false)
new_instance.persisted? #=> true

another_instance = Example.new
another_instance.name = ""
another_instance.save(validate: true)
another_instance.persisted? #=> false
```

No breaking changes and works as well with `#save!`

### CODE REVIEW

**Comment by Blacksmoke16** [comment]
What's the use case for this?

**Comment by robacarp** [comment]
This pattern is common with ActiveRecord and it's _fine_, but it reveals a misalignment of the way validations are currently implemented with the real state of the way applications are built. It is often true that complex models end up with several states of validity: a value can be blank except when some other value is true; or validations need only be applied when a final state is reached; etc. 

But I think this change is a relevant way to extend the current model so that it can be used in the various circumstances encountered with building applications.

File: src/granite/transactions.cr [hunk 1]
```diff
@@ -169,12 +169,12 @@ module Granite::Transactions
   # The save method will check to see if the primary exists yet. If it does it
   # will call the update method, otherwise it will call the create method.
   # This will update the timestamps appropriately.
-  def save
+  def save(**args)
```
Comment (Blacksmoke16) [inline src/granite/transactions.cr:172]:
```suggestion
  def save(*, validate : Bool = true)
```

File: src/granite/transactions.cr [hunk 2]
```diff
@@ -169,12 +169,12 @@ module Granite::Transactions
...
     {% begin %}
     {% primary_key = @type.instance_vars.find { |ivar| (ann = ivar.annotation(Granite::Column)) && ann[:primary] } %}
     {% raise raise "A primary key must be defined for #{@type.name}." unless primary_key %}
     {% ann = primary_key.annotation(Granite::Column) %}
-    return false unless valid?
+    return false if args.fetch("validate", true) && !valid?
```
Comment (Blacksmoke16) [inline src/granite/transactions.cr:177]:
```suggestion
    return false if validate && !valid?
```

File: src/granite/transactions.cr [hunk 3]
```diff
@@ -199,8 +199,8 @@ module Granite::Transactions
   {% end %}
   end
 
-  def save!
-    save || raise Granite::RecordNotSaved.new(self.class.name, self)
+  def save!(**args)
+    save(**args) || raise Granite::RecordNotSaved.new(self.class.name, self)
```
Comment (Blacksmoke16) [inline src/granite/transactions.cr:203]:
```suggestion
  def save!(*, validate : Bool = true)
    save(validate: validate) || raise Granite::RecordNotSaved.new(self.class.name, self)
```

**Review by eliasjpr — APPROVED** [review]

**Pull request merged**